File: VB/PageMerging/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI

Namespace PageMerging
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			Dim report As New MainReport()
			Dim ds As New DataSet1()
			Dim rnd As New Random(DateTime.Now.Millisecond)
			For i As Integer = 0 To 49
				Dim OrderID As Integer = rnd.Next(100)
				Dim ProductID As Integer = rnd.Next(50)
				If ds.Order_Details.FindByOrderIDProductID(OrderID, ProductID) Is Nothing Then
					ds.Order_Details.AddOrder_DetailsRow(OrderID, ProductID, CDec(Math.Round(rnd.NextDouble()*50, 2)), 5, 0.0F)
				End If
			Next i
			report.DataSource = ds
			AddHandler report.AfterPrint, AddressOf report_AfterPrint
			report.ShowPreview()
		End Sub

		Private Sub report_AfterPrint(ByVal sender As Object, ByVal e As EventArgs)
			Dim add As New AdditionalReport()
			add.CreateDocument()
			DirectCast(sender, MainReport).Pages.AddRange(add.Pages)
		End Sub
	End Class
End Namespace

File: CS/PageMerging/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;

namespace PageMerging {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }
        private void simpleButton1_Click(object sender, EventArgs e) {
            XtraReport mainReport = CreateMainReport();                       
            mainReport.ShowPreview();
        }

        private void MainReport_AfterPrint(object sender, EventArgs e) {
            XtraReport additionalReport = CreateAdditionalReport();
            (sender as XtraReport).ModifyDocument(x => { x.AddPages(additionalReport.Pages); });
        }

        private XtraReport CreateMainReport() {
            MainReport mainReport = new MainReport();
            mainReport.AfterPrint += MainReport_AfterPrint;
            DataSet1 ds = new DataSet1();
            Random rnd = new Random(DateTime.Now.Millisecond);
            for(int i = 0; i < 50; i++) {
                int OrderID = rnd.Next(100);
                int ProductID = rnd.Next(50);
                if(ds.Order_Details.FindByOrderIDProductID(OrderID, ProductID) == null)
                    ds.Order_Details.AddOrder_DetailsRow(OrderID, ProductID, (decimal)Math.Round(rnd.NextDouble() * 50, 2), 5, 0.0f);
            }
            mainReport.DataSource = ds;
            mainReport.CreateDocument();
            return mainReport;
        }

        private XtraReport CreateAdditionalReport() {
            AdditionalReport add = new AdditionalReport();
            add.CreateDocument();
            return add;
        }        
    }
}